File: src/main/java/com/redbox/domain/notice/dto/NoticeListResponse.java
package com.redbox.domain.notice.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
public class NoticeListResponse {

    private Long noticeNo;
    private String title;
    private LocalDate createdDate;
    private String writer;
    private int views;
    private boolean hasAttachFiles;

    public NoticeListResponse(Long noticeNo, String title, LocalDateTime createdDate, String writer, int views, boolean hasAttachFiles) {
        this.noticeNo = noticeNo;
        this.title = title;
        this.createdDate = createdDate.toLocalDate();
        this.writer = writer;
        this.views = views;
        this.hasAttachFiles = hasAttachFiles;
    }
}

File: migration/src/main/kotlin/com/redbox/domain/community/notice/dto/NoticeListResponse.kt
package com.redbox.domain.community.notice.dto

import java.time.LocalDate
import java.time.LocalDateTime

data class NoticeListResponse(
    val noticeNo: Long,
    val title: String,
    val createdDate: LocalDate,
    val writer: String,
    val views: Int,
    val hasAttachFiles: Boolean
) {
    constructor(noticeNo: Long, title: String, createdDate: LocalDateTime, writer: String, views: Int, hasAttachFiles: Boolean) : this(
        noticeNo,
        title,
        createdDate.toLocalDate(),
        writer,
        views,
        hasAttachFiles
    )
}